kanalony-storage-logic-core/src/main/scala/kanalony/storage/logic/QueryBase.scala

```diff
@@ -38,8 +38,17 @@ abstract class QueryBase[TReq, TQueryRow] extends IQuery {
     }
   }
 
-  def getGroupAggregatedValue(v: List[List[String]]): Double = {
-    v.map(row => row(metricValueLocationIndex).toDouble).sum
+  def getGroupAggregatedValue(v: List[List[String]], metric : Int): Double = {
+    val metricKind = InternalMetrics.values.find(_.id == metric)
```

Comment:
Or maybe you are being defensive in case the metric is not defined on the enum...?